src/api/services/releaseService.ts

```diff
@@ -2,19 +2,190 @@ import { http } from '../index';
-import { Release } from '../../models/release';
+import { Release, ReleaseAnimation } from '../../models/release';
 import { ApiUrls } from '../../utils/types/apiUrls';
+import { ReleaseAnimationType } from '../../enums/releaseAnimationType';
+import { ReleaseVoiceStatus } from '../../enums/releaseVoiceStatus';
+import { ReleaseType } from '../../enums/releaseType';
 
-/**
- * Fetches full releases.
- * @param vnId Visual novel id.
- */
-export const fetchFullReleases = async(vnId: string): Promise<Release[]> => {
-  const { data } = await http.post<DataWrapper<ReleaseDto>>(
-    ApiUrls.Vndb,
-    `get release basic,details,producers (vn = ${vnId})`,
-  );
-
-  const releases = data.data.items.map(dto => releaseFromDto(dto));
-  return releases;
-};
+interface ReleaseIcon {
+
+  /**
+   * Label of release icon.
+   */
+  label: string;
+
+  /**
+   * Name of release icon.
+   */
+  icon: string;
+}
+
+export namespace ReleaseService {
+
+  /**
+   * Fetches full releases.
+   * @param vnId Visual novel id.
+   */
+  export const fetchFullReleases = async(vnId: string): Promise<Release[]> => {
+    let currentPage = 1;
+    let hasToFetchMore = true;
+    const releasesChunks = [];
+
+    /**
+     * Fetches releases by page.
+     * @param page Page.
+     */
+    const fetch = async(page: number): Promise<DataWrapper<ReleaseDto>> => {
+      const { data } = await http.post<DataWrapper<ReleaseDto>>(
+        ApiUrls.Vndb,
+        `get release basic,details,producers (vn = ${vnId}) {"results": 25, "page": ${page}, "sort": "released"}`,
+      );
+
+      return data;
+    };
+
+    while (hasToFetchMore) {
+    // eslint-disable-next-line no-await-in-loop
+      const releasesData = await fetch(currentPage);
+      releasesChunks.push(releasesData.data.items);
+
+      if (releasesData.data.more) {
+        currentPage += 1;
+      } else {
+        hasToFetchMore = false;
+      }
+    }
+
+    return releasesChunks.map(releasesChunk => releasesChunk.map(dto => releaseFromDto(dto))).flat();
+  };
+
+  /**
+   * Gets free/non-free icon for release.
+   * @param isFreeware Whether release if free or not.
+   */
+  export const getIsFreewareIcon = (isFreeware: boolean): ReleaseIcon => {
+    if (isFreeware) {
+      return {
+        label: 'Freeware',
+        icon: 'eva:gift-fill',
+      };
+    }
+
+    return {
+      label: 'Non-free',
+      icon: 'bx:bx-dollar',
+    };
+  };
+
+  /**
+   * Gets icon for release status.
+   * @param releaseType Release type.
+   */
+  export const getReleaseStatusIcon = (releaseType: ReleaseType): ReleaseIcon => {
+    const label = ReleaseType.toReadable(releaseType);
+    const icon = (() => {
+      switch (releaseType) {
+        case ReleaseType.Trial:
+          return 'carbon:circle-dash';
+        case ReleaseType.Partial:
+          return 'carbon:incomplete';
+        default:
+          return 'carbon:circle-solid';
+      }
+    })();
```

Comment:
I think it's a bit cleaner.
```suggestion
  const icon = {
    [ReleaseType.Trial]: 'carbon:circle-dash',
    [ReleaseType.Partial]: 'carbon:incomplete',
    [ReleaseType.Complete]: 'carbon:circle-solid',
  }[releaseType];
```